File: typescript/session2/heritageMultiple/NatParIntRecursif.ts
import {IntPositif} from "./IntPositif";
import {AlgebreNatRecursif} from "./AlgebreNatRecursif";
import {Nat} from "../../session1/td/Nat/Nat";

export class NatParIntRecursif extends IntPositif, AlgebreNatRecursif {

    constructor(rep : number) {
        super(rep);
    }

    creerNatAvecValeur(val : number) : Nat{
        return new NatParIntRecursif(val);
    }

    toString() : string {
        return '' + this.getVal();
    }
    
    equals(obj : Object) : boolean {
        if(!(obj instanceof Nat))
            return false;
        let x = <Nat>obj;
        return this.getVal() == x.getVal();
    }
}

File: java/cours/src/session2/td/heritageMultiple/NatParIntRecursif.java
package session2.td.heritageMultiple;

import session1.td.Nat;

public class NatParIntRecursif extends IntPositif implements AlgebreNatRecursif {

	public NatParIntRecursif(int rep) {
		super(rep);
	}
	
	@Override
	public Nat creerNatAvecValeur(int val) {
		return new NatParIntRecursif(val);
	}
	
	@Override
	public String toString() {
		return Integer.toString(this.val());
	}
	@Override
	public boolean equals(Object obj) {
		if(!(obj instanceof Nat))
			return false;
		Nat x = (Nat)obj;
		return this.val() == x.val();
	}
}
